validators/item-validator.js

```diff
@@ -0,0 +1,35 @@
+const { body, validationResult } = require("express-validator");
+const Category = require("../models/category");
+
+exports.generateValidator = [
+  body("name", "Name must not be empty.").trim().isLength({ min: 1 }).escape(),
+  body("description", "Description must not be empty.")
+    .trim()
+    .isLength({ min: 1 })
+    .escape(),
+  body("price", "Only decimals allowed")
+    .trim()
+    .optional({ checkFalsy: true })
+    .isNumeric(),
+  body("number_in_stock", "Only decimals allowed")
+    .trim()
+    .optional({ checkFalsy: true })
+    .isNumeric(),
+  (req, res, next) => {
```

Comment:
Please add this to the controller layer, we should only have the validation logic in the validator layer.